py/torch_tensorrt/dynamo/conversion/impl/pool.py

```diff
@@ -104,3 +106,53 @@ def max_poolNd(
 
     set_layer_name(pool_layer, target, name, source_ir)
     return pool_layer.get_output(0)
+
+
+def adaptive_avg_pool1d(
+    ctx: ConversionContext,
+    target: Union[Target, str],
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    output_size: Union[int, Sequence[int]],
+) -> TRTTensor:
+    def start_index(idx: int, out_dim: int, in_dim: int) -> int:
+        return math.floor((float(idx) * float(in_dim)) / out_dim)
+
+    def end_index(idx: int, out_dim: int, in_dim: int) -> int:
+        return math.ceil((float(idx + 1) * float(in_dim)) / out_dim)
+
+    in_dim = input.shape[-1]
+    out_dim = output_size if isinstance(output_size, int) else output_size[0]
+    output_list = []
+
+    for i in range(out_dim):
+        start = start_index(i, out_dim, in_dim)
+        end = end_index(i, out_dim, in_dim)
+
+        slices = []
+        for j in range(start, end):
+            slice = impl.select.select(
+                ctx, target, source_ir, f"{name}_select_{i}_{j}", input, -1, j
+            )
```

Comment:
Is this implementing the operation `input[i:j]`? Is a select on `i` also needed?

Reply:
Actually, this is `input[start:end]` where `start` and `end` are calculated by `i` and `start_index` and `end_index` functions. I think the updated comments will make this readable.

Reply:
- Consider case where multiple `i` have same `j` - repeated slice